common/onepanel/base/vars.yaml

```diff
@@ -19,4 +19,11 @@ database:
     default: onepanel
   driverName:
     required: true
-    default: postgres
\ No newline at end of file
+    default: postgres
+application:
+  webHost:
```

Comment:
let's update this section and its default like so:
```
application:
  host: <host-or-ip>
  apiPort: 32001
  uiPort: 32000
```
Do we have a way to change the key/value here based on provider, for example, if it's local, it'll be `uiPort`/`apiPort`, but if it's GKE, AKS or EKS it'll be `uiPath`, `apiPath`.

Reply:
The CLI command `init` updates the "params.yaml" file.
So I think we can.
We just need to know the provider ahead of time.
I'm thinking they have to pass the provider in?
`main init --provider=microk8s`?

And we output something like...
"Running init with provider microk8s..."

In manifests, we'd need to put placeholders for all the providers.

Reply:
Hmm, but how do we instruct GKE, AKS, to use those values?
Would `main deployment` over-write the deployment files based on provider?

Reply:
In case of the cloud providers, we deploy istio and virtual services with those paths.

Reply:
Need to clarify something here, whatever they put in for `uiPort`, `apiPort` needs to persist in the YAML. I'm not exactly sure if it is or it's random. Also, because we need both grpc and http for api, this is the `params.yaml` structure:
```
application:
  host: <host-or-ip>
  apiHTTPPort: 32000
  apiGRPCPort: 32001
  uiPort: 32002
```

Reply:
Just confirmed that the nodePort is random, so it needs to be set based on their input above, example:
```
spec:
  type: NodePort
  ports:
    - port: 8888
      targetPort: 8888
      nodePort: < apiHTTPPort>
      name: "http"
    - port: 8887
      targetPort: 8887
      nodePort: < apiGRPCPort>
      name: "grpc"
```

Reply:
@rushtehrani @aleksandrmelnikov Just joining in here. Why don't we have a set variable for each service? E.g. 
api: http://localhost:8888 or
api: http://www.test.com/api

If we do the split variable approach, we're probably going to have to do some post-processing on the resulting files to replace the appropriate variables.

Reply:
I think this is actually cleaner, the only thing that we need to figure out is the gRPC port since https/http are known but gRPC doesn't have known port as far as I know.

Reply:
@rushtehrani for cloud providers, will it still be a path? 
Maybe we could do
api: http://....
apiGrpc: ....

Reply:
no, it won't be a path. gRPC is transported over http/2 so it's the same scheme, just a different port. I think we can just have a field for `apiGRPCPort` and default it to some port number.

Reply:
Okay, that works.

Reply:
Actually, no wait it doesn't.

If we have api: http://localhost:8888, it's difficult to replace the port with another one.
And if we go the ApiHost with different port keys, we're back to square one. 

We can make gRPC its own key, or we can go back to having several variables and having the CLI do post-processing.

Reply:
Not sure I follow, gRPC is for external consumption, the way these will get setup in Kubernetes is as services. In case of local k8s it'll be `service` with `nodePort` set, in case of cloud providers, it will be `service` and `virtualservice` with paths.